### DISCUSSION TITLE
Send a 204 instead of a 404 when a request for a tile is made outside the extent of the COG

### DISCUSSION THREAD

**Discussion by devdattaT** [developmentseed/titiler#742] [General]
I am building an app where the frontend is in MapboxGL and it is consuming tiles generated on the fly by TiTiler.

As the user navigates around the map, the client library makes tiles request which could fall outside the area of the Raster. Right now, TiTiler sends a 404 Error message, and this is also shown as an error by MapboxGL.

Given that there is no content to be returned a 204 HTTP status code might make more sense. This is what is also suggested on the MapBoxGL repo: https://github.com/mapbox/mapbox-gl-js/issues/1800

**Comment by vincentsarago** [comment]
@devdattaT that's a bit tricky but hopeful it's really easy for people to customized the error message return by their titiler application. 

People could argue that tile outside a bound is not found not empty 🤷‍♂️

Ideally you'll want to configure your raster layer so it doesn't try to load tiles outside the raster bounding box. 

https://github.com/developmentseed/titiler/blob/b66f228c7809b8706addf22508d7d4589ecfdb15/src/titiler/core/titiler/core/errors.py#L35
